Xamarin.Forms.Platform.UAP/FontExtensions.cs

```diff
@@ -109,12 +109,26 @@ public static FontFamily ToFontFamily(this string fontFamily)
 
 		static string FindFontFamilyName(string fontFile)
 		{
-			using (var fontSet = new CanvasFontSet(new Uri(fontFile)))
+			try
 			{
-				if (fontSet.Fonts.Count == 0)
-					return null;
+				var fontUri = new Uri(fontFile, UriKind.RelativeOrAbsolute);
 
-				return fontSet.GetPropertyValues(CanvasFontPropertyIdentifier.FamilyName).FirstOrDefault().Value;
+				// CanvasFontSet only supports ms-appx:// and ms-appdata:// font URIs
+				if (fontUri.IsAbsoluteUri && (fontUri.Scheme == "ms-appx" || fontUri.Scheme == "ms-appdata"))
+				{
+					using (var fontSet = new CanvasFontSet(fontUri))
+					{
+						if (fontSet.Fonts.Count != 0) 
+							return fontSet.GetPropertyValues(CanvasFontPropertyIdentifier.FamilyName).FirstOrDefault().Value;
+					}
+				}
+
+				return null;
+			}
+			catch
+			{
+				// the CanvasFontSet constructor can throw an exception in case something's wrong with the font. It should not crash the app
```

Comment:
hmmm so there's really know way to check if this exists? Just have to catch the exception?

Reply:
It's not just about checking if it exists. The font could be corrupt and crash the app if the exception is not handled. The check for a URI with the proper scheme is already done to prevent an exception, but there's no clean way to guard against invalid font files.